aws-xray-recorder-sdk-spring/src/main/java/com/amazonaws/xray/spring/aop/AbstractXRayInterceptor.java

```diff
@@ -0,0 +1,102 @@
+package com.amazonaws.xray.spring.aop;
+
+import com.amazonaws.xray.AWSXRay;
+import com.amazonaws.xray.entities.Subsegment;
+import com.amazonaws.xray.exceptions.SegmentNotFoundException;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Pointcut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class AbstractXRayInterceptor {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(this.getClass());
+
+    protected Object processXRayTrace(ProceedingJoinPoint pjp) throws Throwable {
+        boolean endSegment = false;
+        try {
+            if (AWSXRay.getCurrentSegment() != null) {
+                LOGGER.trace("Current segment exists");
+            }
+        } catch (SegmentNotFoundException snfe) {
```

Comment:
The segment not found cases should be passed to the current recorder's `ContextMissingStrategy`. We try to protect users from inadvertent tracing context loss (happens most often when trying to trace multi-threaded operations). In order to protect users from losing their tracing context, we must by default throw an exception when a user tries to create a subsegment with no segment in progress. Of course, users may override this if they desire. But the default behavior should pass the exception onto the context missing strategy and allow that strategy to create a default segment, if it so desires.